Do not link nextbus coordinator to config entry

epenet opened the pull request:
## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Spotted in #128148

The coordinator is shared among all `nextbus` config entries

This fixes a bug where unloading one nextbus config entry would break the shared coordinator

Other potentially unwanted side effects:
- The `pref_disable_polling` option of the first loaded config entry would take precedence.
- Background tasks would get created against the first loaded config entry
- Reauth would trigger against the first loaded config entry

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] I have followed the [perfect PR recommendations][perfect-pr]
- [ ] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

epenet committed:
Do not link nextbus coordinator to config entry

emontnemery commented on homeassistant/components/nextbus/__init__.py:
```diff
@@ -27,7 +27,9 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
-    await coordinator.async_config_entry_first_refresh()
+    await coordinator.async_refresh()
+    if not coordinator.last_update_success:
+        return False
```
This doesn't seem right, doesn't returning `False` here mean the config entry setup permanently fails, unlike if we raise `ConfigEntryNotReady`?

epenet replied:
> This doesn't seem right, doesn't returning `False` here mean the config entry setup permanently fails, unlike if we check for exceptions and raise `ConfigEntryNotReady`?

Ah indeed!
I have fixed this, and added specific tests

epenet committed:
Refactor tests and add specific failure test

epenet committed:
Use ConfigEntryNotReady

ViViDboarder commented:
Does this actually occur? I added a test and a patch to prevent this in my last PR.

epenet committed:
Cleanup coordinator

epenet commented:
> Does this actually occur? I added a test and a patch to prevent this in my last PR.

Ha! I didn't spot the workaround from #126161.
I have removed the workaround as it should not be needed anymore since the `config_entry` is no longer set.

Also, you should not have been calling `async_config_entry_first_refresh` more than once.
Despite #126161 you would end-up calling it multiple times (once for each entry).

I think it's better to keep it fully independant from any config entry.

ViViDboarder commented:
Awesome! Thanks. I was wondering how I missed not just setting `config_entry` to `None` in the super init, but then I saw that it wasn't an option until 2 days ago!

ViViDboarder reviewed (approved):
Looks good. Thanks. I'm glad to have a less hacky way of preventing unloading.

gjohansson-ST reviewed (approved):
Thanks @epenet 👍
